packages/@aws-cdk/aws-elasticsearch/lib/domain.ts

```diff
@@ -1167,7 +1227,33 @@ export class Domain extends DomainBase implements IDomain {
       throw new Error(`Unknown Elasticsearch version: ${elasticsearchVersion}`);
     }
 
-    const encryptionAtRestEnabled = props.encryptionAtRest?.enabled ?? (props.encryptionAtRest?.kmsKey != null);
+    const masterUserArn = props.advancedSecurity?.masterUserArn;
+    const masterUserName = props.advancedSecurity?.masterUserName;
+
+    const advancedSecurityEnabled = (masterUserArn ?? masterUserName) != null;
+    const internalUserDatabaseEnabled = masterUserName != null;
+    const masterUserPassword =
+      props.advancedSecurity?.masterUserPasswordSecret?.toString() ??
+      internalUserDatabaseEnabled
+        ? new secretsmanager.Secret(this, id, {
+          generateSecretString: {
+            secretStringTemplate: JSON.stringify({
+              username: masterUserName,
+            }),
+            generateStringKey: 'password',
+          },
+        })
+          .secretValueFromJson('password')
+          .toString()
+        : undefined;
+
+    const encryptionAtRestEnabled = advancedSecurityEnabled
+      ? true
+      : props.encryptionAtRest?.enabled ??
+        props.encryptionAtRest?.kmsKey != null;
+    const nodeToNodeEncryptionEnabled = advancedSecurityEnabled
```

Comment:
Same question?

Reply:
Both are required if you're using Advanced Security!

Reply:
My thought was that we set these for the user so that they can't create an invalid configuration. Thoughts?

Reply:
TBH I find it a strange user experience on the cloud formation side. The name `AdvancedSecurityOptions` doesn't match what the configuration is for.

> Specifies options for fine-grained access control.

If I look at that and the settings I would not assume that this has anything to do with node to node encryption or encryption at rest.

Given that the default is to not have encryption at rest and node to node encryption my preference would be to error if advanced security is set and these are not set so the user is aware of what is happening. What do you think? @iliapolo do you have any thoughts?